src/Data/Aeson/Config/FromValue.hs

```diff
@@ -143,6 +147,9 @@ instance (Selector sel, FromValue a) => GenericDecode (S1 sel (Rec0 a)) where
 instance {-# OVERLAPPING #-} (Selector sel, FromValue a) => GenericDecode (S1 sel (Rec0 (Maybe a))) where
   genericDecode = accessFieldWith (.:?)
 
+instance {-# OVERLAPPING #-} (Selector sel, FromValue a) => GenericDecode (S1 sel (Rec0 (Last a))) where
+  genericDecode = accessFieldWith (fmap Last .* (.:?))
```

Comment:
I think
```haskell
  genericDecode = accessFieldWith (\ value key -> Last <$> (value .:? key))
```
is easier to understand.

I like the use of `Last`, but if you want to add the instance you also need to add tests (similar to what I just did for `Maybe` in https://github.com/sol/hpack/commit/35902395c0dbe2a96b678eef2f93168cd39b80d2, don't add the nested instance, unless we are going to use it though).

Alternatively use `<|>` (as we currently do in other places).

Reply:
Noted. I will also retain `Last` and add tests.